src/overlays/actors/ovl_Bg_Ice_Turara/z_bg_ice_turara.c

```diff
@@ -27,23 +41,164 @@ const ActorInit Bg_Ice_Turara_InitVars = {
     (ActorFunc)BgIceTurara_Update,
     (ActorFunc)BgIceTurara_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/BgIceTurara_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ice_Turara/BgIceTurara_Destroy.s")
+InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneScale, 600, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, -3, ICHAIN_CONTINUE),
+    ICHAIN_F32(minVelocityY, -30, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+// gravitational acceleration
+Vec3f D_8089267C[] = { 0.0f, -1.0f, 0.0f };
+
+Color_RGBA8 primColor[] = { 170, 255, 255, 255 };
+
+Color_RGBA8 envColor[] = { 0, 50, 100, 255 };
+
+void BgIceTurara_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIceTurara* this = THIS;
+    s16 pad;
+    ColHeader* colHeader = NULL;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_06002594, &colHeader);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, thisx, &sCylinderInit);
+    Collider_CylinderUpdate(thisx, &this->collider);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+    if (thisx->params == 0) {
+        this->actionFunc = func_80892220;
+        return;
+    }
+    this->dyna.actor.shape.rot.x = -0x8000;
+    this->dyna.actor.shape.unk_08 = 1200.0f;
+    this->actionFunc = func_80892280;
+}
+
+void BgIceTurara_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIceTurara* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgIceTurara_Break(BgIceTurara* this, GlobalContext* globalCtx, f32 arg2) {
+    Vec3f velvec;
+    Vec3f posvec;
+    s32 phi_s0;
+    s32 sp88;
+
+    Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.posRot.pos, 30, NA_SE_EV_ICE_BROKEN);
+
+    for (sp88 = 0; sp88 != 2; sp88++) {
+        for (phi_s0 = 0; phi_s0 != 10; phi_s0++) {
+
+            posvec.x = this->dyna.actor.posRot.pos.x + Math_Rand_CenteredFloat(8.0f);
+            posvec.y = this->dyna.actor.posRot.pos.y + (Math_Rand_ZeroOne() * arg2) + (sp88 * arg2);
+            posvec.z = this->dyna.actor.posRot.pos.z + Math_Rand_CenteredFloat(8.0f);
+
+            velvec.x = Math_Rand_CenteredFloat(7.0f);
+            velvec.z = Math_Rand_CenteredFloat(7.0f);
+            velvec.y = (Math_Rand_ZeroOne() * 4.0f) + 8.0f;
+
+            EffectSsEnIce_Spawn(globalCtx, &posvec, (Math_Rand_ZeroOne() * 0.2f) + 0.1f, &velvec, D_8089267C,
+                                primColor, envColor, 30);
+        }
+    }
+}
+
+void func_80892220(BgIceTurara* this, GlobalContext* globalCtx) {
+    if ((this->collider.base.acFlags & 2) != 0) {
```

Comment:
```suggestion
    if (this->collider.base.acFlags & 2) {
```
These sorts of flag comparisons are boolean, so it's OK to leave out `!= 0`.